pkg/controllers/provisioning/provisioner.go

```diff
@@ -285,35 +278,27 @@ func (p *Provisioner) schedule(ctx context.Context, pods []*v1.Pod, stateNodes [
 	return nodes, err
 }
 
-func (p *Provisioner) launch(ctx context.Context, opts LaunchOptions, node *scheduler.Node) (string, error) {
+func (p *Provisioner) launch(ctx context.Context, scheduledMachine *scheduler.Machine, opts ...LaunchOption) (string, error) {
 	// Check limits
 	latest := &v1alpha5.Provisioner{}
-	name := node.Requirements.Get(v1alpha5.ProvisionerNameLabelKey).Values()[0]
-	if err := p.kubeClient.Get(ctx, types.NamespacedName{Name: name}, latest); err != nil {
+	if err := p.kubeClient.Get(ctx, types.NamespacedName{Name: scheduledMachine.ProvisionerName}, latest); err != nil {
```

Comment:
Should this be an ignorenotfound?

Reply:
No, it shouldn't be. We should error on this and return the error up the stack so that we know that we couldn't find the provisioner of the machine that we intended to launch.